### DISCUSSION TITLE
iPerf3 loss counters sender / receiver

### DISCUSSION THREAD

**Discussion by dbillon** [esnet/iperf#1639] [Q&A]
Hi,

I'm trying to measure the bandwidth available between a Macbook Air M2 (MacOS 14.3 - iPerf V3.16) and a Raspberry PI 4 (Debian 12 - iPerf V3.12)
There a wifi network between those 2 boxes ... some drops are expected above a few hundreds of Mb/s.

I can't explained why I'm not seeing those drops reported at the end of the test by the receiver:

```
❯ iperf3 -c 192.168.1.206 -u -b 300M
Connecting to host 192.168.1.206, port 5201
[  5] local 192.168.1.222 port 59643 connected to 192.168.1.206 port 5201
[ ID] Interval           Transfer     Bitrate         Total Datagrams
[  5]   0.00-1.00   sec  35.7 MBytes   300 Mbits/sec  25880
[  5]   1.00-2.00   sec  35.8 MBytes   300 Mbits/sec  25898
[  5]   2.00-3.00   sec  35.8 MBytes   300 Mbits/sec  25897
[  5]   3.00-4.00   sec  35.8 MBytes   300 Mbits/sec  25900
[  5]   4.00-5.00   sec  35.8 MBytes   300 Mbits/sec  25898
[  5]   5.00-6.00   sec  35.8 MBytes   300 Mbits/sec  25896
[  5]   6.00-7.00   sec  35.8 MBytes   300 Mbits/sec  25898
[  5]   7.00-8.00   sec  35.8 MBytes   300 Mbits/sec  25896
[  5]   8.00-9.00   sec  35.8 MBytes   300 Mbits/sec  25900
[  5]   9.00-10.00  sec  35.8 MBytes   300 Mbits/sec  25897
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bitrate         Jitter    Lost/Total Datagrams
[  5]   0.00-10.00  sec   358 MBytes   300 Mbits/sec  0.000 ms  0/258960 (0%)  sender
[  5]   0.00-10.12  sec   285 MBytes   236 Mbits/sec  0.059 ms  0/258957 (0%)  receiver

```

From the last line in the report, it's obvious that the receiver did not receive all datagrams.
But it still shows 0% loss.

Any idea ?

BR,

Damien.

**Comment by davidBar-On** [comment]
Hi Damien,
Note that per the receiver's Total number of packers received, no packet was lost (the 3 missing packets are probably packets that were still on the way to the server).  Also, the received data is almost exactly 80% of the sent data.  Therefore, it may be that for some reason, the server receives/read only 80% of each UDP packet.  The following may help with evaluating the problem, especially whether the packets are truncated somewhere:
1. UDP packets segmentation may cause such issue.  Try running iperf3 with `-l 512` which is the size that all networks should support.
2. Determine the size of the UDP packets sent using `-V` (" UDP block size").  Then determine the actual packets size sent and received using tools such as tcpdump or Wireshark - on both the client and the server sides.
3. If the above step does **not** show that the packets were truncated, use `-d` option on the server side to see the size of the packets that the server reads.
4. Try running in reverse mode (`-R`) to see if the issue occurs on both directions.
